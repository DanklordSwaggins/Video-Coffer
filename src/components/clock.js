import React from "react";
import moment from "moment";
import "../styles/clock.css";

class Clock extends React.Component {
  constructor(props) {
    super(props);
    this.state = { time: new moment().format("LLL").toLocaleString() };
  }

  tick() {
    this.setState({ time: new moment().format("LLL").toLocaleString() });
  }

  componentDidMount() {
    this.intervalID = setInterval(() => this.tick(), 1000);
  }

  componentWillUnmount() {
    clearInterval(this.intervalID);
  }

  render() {
    return <h6 className="clock">{this.state.time}</h6>;
  }
}

export default Clock;
