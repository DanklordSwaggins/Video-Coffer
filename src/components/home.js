import React from 'react';
import '../styles/home.css';
import SearchBar from './searchBar';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';
import {history} from '../routes';

const toplayer = ()=>{
        //ReactDOM.render(jsx, document.getElementById('root'));
        history.push('/videoPlayer')
};

export const Home = ({startLogout}) => {
    return (
        <div className="Home">
            <div className="lander">
                <h1>Video Coffer</h1>
                <p>If you love a video, don't let it go</p>
                <SearchBar />
                <div>
                    <button onClick={function(event){ startLogout()}}>Logout</button>
                    <button onClick={function(event){ toplayer()}}>player</button>
                </div>
            </div>
        </div>
    );
};

const mapDispatchToProps=(dispatch)=>({
    startLogout: () =>dispatch(startLogout())
});

export default connect(undefined,mapDispatchToProps)(Home);