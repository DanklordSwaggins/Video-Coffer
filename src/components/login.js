import React from 'react';
import { connect } from 'react-redux';
import { Button } from 'react-bootstrap';
import { startLogin } from '../actions/auth';
import '../styles/login.css';

export const LoginPage = ({ startLogin }) => {

  return (
    <div className="Login">
      <Button
        variant="danger"
        onClick={startLogin}
      >
        Sign in with Google
      </Button>
    </div>
  );
};


const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});



export default connect(undefined, mapDispatchToProps)(LoginPage);
