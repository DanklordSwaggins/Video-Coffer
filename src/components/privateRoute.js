import React, {useEffect, useState} from 'react';
import {firebase} from '../firebase/firebaseConfig';


// const privateRoute = props => {
//     const [authenticated, setAuthenticatedStatus] = 
//     useEffect(() => {
//         firebase.auth().onAuthStateChanged((user)=>{ 
//             if(user){
//               //console.log('logged in')
//               //window.location.replace('/home');
//               // set state
//               history.push('/home');
//             }else{
//               history.push('/');
//             }
//           });
//     }, [])

//     return (

//     )
// }
