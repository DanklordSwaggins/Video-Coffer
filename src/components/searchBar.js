import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { InputGroup, FormControl, Spinner, Card, Button, CardDeck } from 'react-bootstrap';
import { debounce } from 'throttle-debounce';
import videoSearch from '../actions/videos';
import videoId from '../actions/videoId';
import '../styles/searchBar.css';


const SearchBar = () => {

    const dispatch = useDispatch();
    const videos = useSelector(state => state.videos);
    const { fetched, fetching, error } = videos;

    let history = useHistory();

    const handleOnChange = debounce(700, (e) => {
        dispatch(videoSearch({ keywords: e, maxResults: 50, pageToken: '' }))
    });

    const handleClick = (key) => {
        dispatch(videoId(key))
        history.push('/videoPlayer')
    };

    return (
        <div>
            <InputGroup size="lg">
                <FormControl
                    placeholder="Inject queries here..."
                    onChange={(value) => handleOnChange(value.target.value)}
                />
            </InputGroup>
            <div className="flex-container">
                {fetching &&
                    <Spinner
                        animation="border"
                        variant="danger"
                        role="status"
                    />
                }
                {error ? <p>We encountered an error :(</p> : null}
                {fetched && videos.items.map((item) => {
                    return (
                        <div key={item.id.videoId}>
                            <CardDeck className="results">
                                <Card border="danger" className="cardItem">
                                    <Card.Img variant="top" src={item.snippet.thumbnails.medium.url} />
                                    <Card.Body>
                                        <Card.Title>{item.snippet.title}</Card.Title>
                                        <Card.Text>{item.snippet.description}</Card.Text>
                                    </Card.Body>
                                    <Button
                                        variant="danger"
                                        onClick={() => { handleClick(item.id.videoId) }}
                                    >Play</Button>
                                    <Button variant="warning">Watch Later</Button>
                                </Card>
                            </CardDeck>
                        </div>
                    )
                })}
            </div>
        </div>
    );
};

export default SearchBar;