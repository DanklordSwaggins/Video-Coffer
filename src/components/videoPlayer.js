import React from 'react';
import { useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { Button } from 'react-bootstrap';
import '../styles/videoPlayer.css';

const VideoPlayer = () => {

    const videoId = useSelector(state => state.videoId);
    const { id } = videoId;

    let history = useHistory();

    const handleClick = () => {
        history.push('/home');
    }

    return (
        <div className="container">
            <div className="mediaPlayer">
                <Button
                    variant="primary"
                    onClick={handleClick}
                >Go Back</Button>
                <iframe
                    src={`https://www.youtube.com/embed/${id}`}
                    frameBorder='0'
                    height='480'
                    width='640'
                    allow='autoplay; encrypted-media'
                    allowFullScreen
                    title='video'
                />
            </div>
        </div>
    );
};

export default VideoPlayer;