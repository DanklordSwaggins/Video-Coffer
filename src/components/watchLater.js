import React from "react";
import "../styles/watchLater.css";
import { useHistory } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { startLogout } from "../actions/auth";
import { removeVideoList } from "../actions/videoId";
import { Card, ButtonGroup, Button, CardDeck } from "react-bootstrap";
import videoId from "../actions/videoId";

export const WatchLater = () => {

  const history = useHistory();
  const dispatch = useDispatch();

  const videolist = useSelector(state => state.list);
  const sName = useSelector(state => state.users);
  const { email } = sName;

  const toPlayer = key => {
    dispatch(videoId(key));
    history.push("/videoPlayer");
  };

  const HandlePush = item => {
    const videoData = {
      sid: item.sid,
      email: email
    };
    dispatch(removeVideoList(videoData));
    history.push("/watchLater");
  };

  return (
    <div className="WatchLater">
      <div className="d-flex flex-column">
        <ButtonGroup>
          <Button variant="warning" onClick={() => history.push("/videoPlayer")}>Video Player</Button>
          <Button variant="warning" onClick={() => history.push("/home")}>Home</Button>
          <Button variant="warning" onClick={() => { dispatch(startLogout()) }}>Logout</Button>
        </ButtonGroup>
      </div>
      <div className="lander">
        <h1>Video Coffer</h1>
        <p>If you love a video, don't let it go</p>
        <div className="flex-container">
          {videolist.map(item => {
            return (
              <div key={item.id}>
                <CardDeck className="results">
                  <Card border="danger" className="cardItem">
                    <Card.Img variant="top" src={item.thumbnail} />
                    <Card.Body>
                      <Card.Title>{item.title}</Card.Title>
                      <Card.Text>{item.description}</Card.Text>
                    </Card.Body>
                    <Button
                      variant="danger"
                      onClick={() => { toPlayer(item.id) }}>Play</Button>
                    <Button
                      variant="warning"
                      onClick={() => { HandlePush(item) }}>Remove from Watch List</Button>
                  </Card>
                </CardDeck>
              </div>
            );
          })}
        </div>
      </div>
    </div>
  );
};

export default WatchLater;