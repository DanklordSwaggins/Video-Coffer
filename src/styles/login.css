@media all and (min-width: 480px) {
    .Login {
      padding: 240px 0;
      margin: auto;
      max-width: 320px;
    }

    .Login Button {
      display: block;
      width: 100%;
      height: 60px;
    }
  }