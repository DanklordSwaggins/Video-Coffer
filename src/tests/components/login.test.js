import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import {LoginPage} from '../../components/login';
import configureStore from '../..//store/store'
import { Provider } from "react-redux";

import {shallow, mount} from 'enzyme';


it('should render login', ()=>{
    const store = configureStore();
    const wrapper = mount(
        <Provider store={store}>
        <LoginPage />
        </Provider>
        
          
    );
    expect(wrapper).toMatchSnapshot();
})
