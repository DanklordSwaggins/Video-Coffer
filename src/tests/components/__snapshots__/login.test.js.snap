// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <LoginPage>
    <div
      className="Login"
    >
      <Button
        active={false}
        disabled={false}
        type="button"
        variant="danger"
      >
        <button
          className="btn btn-danger"
          disabled={false}
          type="button"
        >
          Sign in with Google
        </button>
      </Button>
    </div>
  </LoginPage>
</Provider>
`;
